### PR TITLE
Add a RUN variable to run only specific test files

### PR DESCRIPTION
**Pull request by justinj**
Some small unrelated changes that clippy pointed out.

### CODE REVIEW

**Comment by justinj** [comment]
Would either of @benesch or @mjibson be interested in giving a quick review?

**Review by madelynnblue — COMMENTED** [review]
It looks like the argument to walk can now be a file, is that correct? Should be documented if so.

Also I'm not sure that RUN adds any value here. Kinda bad form for libraries to use env vars. If walk does indeed now support specifying just files, then do we need RUN anymore? The mz tests already allow you to specify a directory, which can now be a file. Am I missing something?

**Comment by justinj** [comment]
Which materialize tests are you referring to? The logic tests don't use datadriven, this is the only way I'm aware of to pass info into a test that can be run via `cargo test`.

**Review by madelynnblue — APPROVED** [review]
I've been convinced. Maybe document in walk that its argument can be a file.

File: src/lib.rs [hunk 1]
```diff
@@ -25,15 +26,30 @@ pub fn walk<F>(dir: &str, mut f: F)
 where
     F: FnMut(&mut TestFile),
 {
+    let mut file_prefix = PathBuf::from(dir);
+    if let Ok(p) = env::var("RUN") {
+        for d in PathBuf::from(&p).iter() {
+            file_prefix.push(d);
+        }
```
Comment (benesch) [inline src/lib.rs]: This looks like `PathBuf::join` to me.

```rust
file_prefix = file_prefix.join(p);
```

File: src/lib.rs [hunk 2]
```diff
@@ -215,14 +229,14 @@ pub struct TestFile {
 }
 
 impl TestFile {
-    fn new(filename: &str) -> Result<Self, Error> {
-        let contents = fs::read_to_string(filename)
-            .with_context(|| format!("error reading file {}", filename))?;
+    fn new(filename: &PathBuf) -> Result<Self, Error> {
+        let contents = fs::read_to_string(filename.to_str().unwrap())
```
Comment (benesch) [inline src/lib.rs]: Just `filename` here works! `read_to_string` takes an `AsRef<Path>`, so converting to a str is unnecessary work.

File: src/lib.rs [hunk 3]
```diff
@@ -215,14 +229,14 @@ pub struct TestFile {
...
+            .with_context(|| format!("error reading file {}", filename.to_str().unwrap()))?;
```
Comment (benesch) [inline src/lib.rs]: The idiom for displaying paths is `filename.display()`, which gives you back a thing that is display-able.

File: src/lib.rs [hunk 4]
```diff
@@ -215,14 +229,14 @@ pub struct TestFile {
...
         let mut res = match Self::parse(&contents) {
             Ok(res) => res,
-            Err(err) => bail!("{}:{}", filename, err),
+            Err(err) => bail!("{}:{}", filename.to_str().unwrap(), err),
```
Comment (benesch) [inline src/lib.rs]: Ditto here.

**Review by benesch — APPROVED** [review]
LGTM! Left you some Rust idiom nits if you like.

**Comment by justinj** [comment]
TFTRs!

**Pull request merged**